megamek/src/megamek/common/actions/WeaponAttackAction.java

```diff
@@ -1303,15 +1296,14 @@ private static String toHitIsImpossible(Game game, Entity ae, int attackerId, Ta
                         && (te != null) && te.hasSeenEntity(ae.getOwner()))
                 && !isArtilleryIndirect && !isIndirect && !isBearingsOnlyMissile) {
             boolean networkSee = false;
-            if (ae.hasC3() || ae.hasC3i() || ae.hasActiveNovaCEWS()) {
+            if (ae.hasC3() || ae.hasC3i() || ae.hasActiveNovaCEWS()
+                && game.getEntitiesVector().stream().anyMatch(en ->
+                    !en.isEnemyOf(ae)
+                    && en.onSameC3NetworkAs(ae)
+                    && Compute.canSee(game, en, target))) {
```

Comment:
Please don't do this inside of a conditional block. If you're going to do a stream, please do so either inside or outside of it.